.github/workflows/test_macos.yaml

```diff
@@ -0,0 +1,81 @@
+name: macOS Test
+
+on: [push, pull_request]
+
+jobs:
+  unit_test:
+    defaults:
+      run:
+        shell: ${{ matrix.run_wrapper || 'bash --noprofile --norc -eo pipefail {0}' }}
+    runs-on: ${{ matrix.runs_on || 'macos-latest' }}
+    strategy:
+      matrix:
+        include:
+          - runs_on: macos-latest
+            python: 3.9
+          - runs_on: apple-silicon-m1
+            run_wrapper: arch -arm64 bash --noprofile --norc -eo pipefail {0}
+            python: 3.9.7
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.x
+        # Needs to be skipped on our self-hosted runners tagged as 'apple-silicon-m1'
+        if: ${{ matrix.runs_on  != 'apple-silicon-m1' }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install Platypus
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          install_platypus
+      - name: Create macOS App Bundle with default options
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          cd osx
+          ./create-osx-bundle.sh
+      - name: Test macOS App Bundle venv
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          pushd ./osx/build/Kivy.app/Contents/Resources/venv/bin
+          source activate
+          popd
+          pip freeze
+          deactivate
+      - name: Test Fix bundle metadata
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          cd osx
+          ./fix-bundle-metadata.sh build/Kivy.app -v "1.2.3"
+          ./fix-bundle-metadata.sh build/Kivy.app -v "master"
+      - name: Test Cleanup
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          cd osx
+          ./cleanup-app.sh build/Kivy.app
+      - name: Test Relocation
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          cd osx
+          ./relocate.sh build/Kivy.app
+      - name: Test dmg creation and store it in osx_artifacts
+        # Needs to be skipped on our self-hosted runners tagged as 'apple-silicon-m1'
+        if: ${{ matrix.runs_on  != 'apple-silicon-m1' }}
+        run: |
+          source ci/osx_ci.sh
+          arm64_set_path_and_python_version ${{ matrix.python }}
+          cd osx
+          ./create-osx-dmg.sh build/Kivy.app Kivy
+          cd ..
+          mkdir osx_artifacts
+          cp osx/Kivy.dmg osx_artifacts/
+      - name: Upload dmg and app as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: KivySDKPackager
+          path: osx_artifacts
```

Comment:
```suggestion
          path: osx_artifacts
      - name: Install Kivy to app
        # Needs to be skipped on our self-hosted runners tagged as 'apple-silicon-m1'
        if: ${{ matrix.runs_on  != 'apple-silicon-m1' }}
        run: |
          source ci/osx_ci.sh
          mount_app
          activate_osx_app_venv
          install_kivy
      - name: Test Kivy install
        # Needs to be skipped on our self-hosted runners tagged as 'apple-silicon-m1'
        if: ${{ matrix.runs_on  != 'apple-silicon-m1' }}
        run: |
          source ci/osx_ci.sh
          activate_osx_app_venv
          test_kivy_install
```

Reply:
Why installing `kivy` again?
What about testing the `kivy` version that has already been packaged and relocated?

Also: We can finally remove the `if: ${{ matrix.runs_on  != 'apple-silicon-m1' }}` check 😃

Reply:
Ooh I forgot that the scripts already include kivy 😬

Nice! What caused the upload to fail before?

Reply:
Actually, we need to install the full and dev deps, which we don't do in the script. Gotta figure out if there's a way to do it without reinstalling kivy.

Reply:
> Ooh I forgot that the scripts already include kivy 😬
> 
> Nice! What caused the upload to fail before?

Looks like `actions/upload-artifact@v2` implementation is in some way latency-sensitive. I've just sent a report of the changes made to our CI runner configuration.

Reply:
> Actually, we need to install the full and dev deps, which we don't do in the script. Gotta figure out if there's a way to do it without reinstalling kivy.

Yep, missed that. 🧐